compose/cli/main.py

```diff
@@ -863,6 +871,22 @@ def up(self, options):
             if cascade_stop:
                 print("Aborting on container exit...")
                 self.project.stop(service_names=service_names, timeout=timeout)
+                if forward_exitval:
+                    def is_us(container):
+                        return container.name_without_project == forward_exitval
+                    candidates = filter(is_us, all_containers)
+                    if not candidates:
+                        log.error('No containers matching the spec "%s" were run.',
+                                  forward_exitval)
```

Comment:
Ideally we want to detect this before running the containers. Unfortunately we don't know how many instances of a certain service will be run, but we should at least check for a service name match before.

Additionally, I think the feature would be improved if the option just took a service name as argument. If the service isn't scaled, the behavior would be the same. If it is, return the first exit code in the list that isn't 0 (or 0 if no container exited abnormally)